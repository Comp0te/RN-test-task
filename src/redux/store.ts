import { createStore, applyMiddleware, combineReducers } from 'redux';
import { combineEpics, createEpicMiddleware } from 'redux-observable';
import { composeWithDevTools } from 'redux-devtools-extension';

import { requestsReducer } from './requests/reducers';

import { requestsEpics } from './requests/epics';

export const rootReducer = combineReducers({
  requests: requestsReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

const rootEpic = combineEpics(
  ...requestsEpics,
);

const epicMiddleware = createEpicMiddleware();

const middleware = [
  epicMiddleware,
];

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(...middleware)),
);

epicMiddleware.run(rootEpic);

export default store;
