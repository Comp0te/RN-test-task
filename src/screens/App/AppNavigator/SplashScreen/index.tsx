import React, { useEffect } from 'react';
import { connect, DispatchProp } from 'react-redux';
import style from './style';

import { SafeAreaView } from 'react-native';

import authService from '../../../../shared/services/auth.service';
import navService from '../../../../shared/services/nav.service';
import * as authAC from '../../../../redux/auth/AC';
import { Spinner } from '../../../../shared/components/Spinner';

type Props = DispatchProp<authAC.Actions>;

const SplashScreen: React.FC<Props> = ({dispatch}) => {

  useEffect(() => {
    const subscriber = authService.getToken().subscribe(token => {
      if (token) {
        dispatch(authAC.Actions.setIsAuthUser(true));
      }

      navService.navigate(token ? 'ProductsNavigator' : 'ProfileNavigator');
      subscriber.unsubscribe();
    });
  }, []);

  return (
    <SafeAreaView style={style.root}>
      <Spinner/>
    </SafeAreaView>
  );
};

export default connect()(SplashScreen);