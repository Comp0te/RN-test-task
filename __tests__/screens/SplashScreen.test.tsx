import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';
import SplashScreen from '../../src/screens/App/AppNavigator/SplashScreen';
import { Provider } from 'react-redux';
import store from '../../src/redux/store';

it('SplashScreen renders correctly', () => {
  expect(renderer.create(
    <Provider store={store}>
      <SplashScreen/>
    </Provider>,
  )).toMatchSnapshot();
});