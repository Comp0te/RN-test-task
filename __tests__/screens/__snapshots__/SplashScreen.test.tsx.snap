// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#860483"
    hidesWhenStopped={true}
    size="large"
  />
</RCTSafeAreaView>
`;
